source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -70,28 +71,26 @@ MStatus MaterialXNode::initialize()
 	MFnStringData stringData;
 
 	MObject theString = stringData.create();
+
 	DOCUMENT_ATTRIBUTE = typedAttr.create(DOCUMENT_ATTRIBUTE_LONG_NAME, DOCUMENT_ATTRIBUTE_SHORT_NAME, MFnData::kString, theString);
-	CHECK_MSTATUS(typedAttr.setStorable(true));
-	CHECK_MSTATUS(typedAttr.setReadable(true));
-	CHECK_MSTATUS(typedAttr.setInternal(true));
-	CHECK_MSTATUS(typedAttr.setCached(false));
-	CHECK_MSTATUS(typedAttr.setHidden(true));
+    CHECK_MSTATUS(typedAttr.setInternal(true));
+    CHECK_MSTATUS(typedAttr.setKeyable(false));
+    CHECK_MSTATUS(typedAttr.setAffectsAppearance(true));
+    CHECK_MSTATUS(typedAttr.setUsedAsFilename(true));
 	CHECK_MSTATUS(addAttribute(DOCUMENT_ATTRIBUTE));
 
 	ELEMENT_ATTRIBUTE = typedAttr.create(ELEMENT_ATTRIBUTE_LONG_NAME, ELEMENT_ATTRIBUTE_SHORT_NAME, MFnData::kString, theString);
-	CHECK_MSTATUS(typedAttr.setStorable(true));
-	CHECK_MSTATUS(typedAttr.setReadable(true));
-	CHECK_MSTATUS(typedAttr.setInternal(true));
-	CHECK_MSTATUS(typedAttr.setCached(false));
-	CHECK_MSTATUS(typedAttr.setHidden(true));
+    CHECK_MSTATUS(typedAttr.setInternal(true));
```

Comment:
Likewise, is "element" a specific enough name?

Reply:
I think "element path" is better as it's not an element it's a path to an element in the document.